master/internal/api_tasks.go

```diff
@@ -219,7 +219,27 @@ func (a *apiServer) GetAllocation(
 	ctx context.Context,
 	req *apiv1.GetAllocationRequest,
 ) (*apiv1.GetAllocationResponse, error) {
-	return nil, grpcutil.UnimplementedError
+	if req.AllocationId == "" {
+		return nil, status.Error(codes.InvalidArgument, "allocation ID missing")
+	}
+
+	resp := model.Allocation{}
+
+	err := db.Bun().NewRaw(`
+	select allocation_id, state, slots, exit_reason, status_code
+	from allocations
+	where allocation_id = ?
+	`, req.AllocationId).Scan(ctx, &resp)
+	if err != nil {
+		return nil, err
+	}
+	return &apiv1.GetAllocationResponse{
+		AllocationId: string(resp.AllocationID),
+		State:        resp.State.Proto(),
+		Slots:        int32(resp.Slots),
+		ExitReason:   &resp.ExitReason,
```

Comment:
```&resp.ExitReason/&resp.StatusCode``` I think will give unexpected results when the allocation has a null exit reason / status code

Here I think we will have a ptr to the zero value rather than a nil ptr

(See other comment)

Reply:
Done!